pkg/metrics/config.go

```diff
@@ -1,11 +1,82 @@
 package metrics
 
+import (
+	"context"
+	"fmt"
+	"github.com/icinga/icinga-go-library/database"
+	schemav1 "github.com/icinga/icinga-kubernetes/pkg/schema/v1"
+	"github.com/pkg/errors"
+	kmetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
 // PrometheusConfig defines Prometheus configuration.
 type PrometheusConfig struct {
-	Url string `yaml:"url"`
+	Url      string `yaml:"url"`
+	Username string `yaml:"username"`
+	Password string `yaml:"password"`
 }
 
 // Validate checks constraints in the supplied Prometheus configuration and returns an error if they are violated.
 func (c *PrometheusConfig) Validate() error {
 	return nil
 }
+
+func SyncPrometheusConfig(ctx context.Context, db *database.DB, config *PrometheusConfig) error {
+	var configPairs []*schemav1.Config
+
+	if config.Url != "" {
+		configPairs = append(configPairs, &schemav1.Config{Key: schemav1.ConfigKeyPrometheusUrl, Value: config.Url})
+		configPairs = append(configPairs, &schemav1.Config{Key: schemav1.ConfigKeyPrometheusLocked, Value: "true"})
+	} else {
+		configPairs = append(configPairs, &schemav1.Config{Key: schemav1.ConfigKeyPrometheusLocked, Value: "false"})
+	}
+
+	stmt, _ := db.BuildUpsertStmt(&schemav1.Config{})
+	if _, err := db.NamedExecContext(ctx, stmt, configPairs); err != nil {
```

Comment:
Do the `INSERT`s/`UPDATE`s in a serializable transaction. Think about why this is a good idea and whether the other side should do that too. See here for how we do it in Icinga DB: https://github.com/Icinga/icingadb/blob/main/pkg/icingadb/ha.go#L289.

As this is a pattern that we are now using a second time, you should consider how this could be done using a function.